geocity/apps/submissions/views.py

```diff
@@ -2373,3 +2437,67 @@ def submission_validations_edit(request, submission_id):
             "formset": formset,
         },
     )
+
+
+@method_decorator(login_required, name="dispatch")
+class ConfirmProlongationTransactionView(View):
+    def get(self, request, pk, prolongation_date, *args, **kwargs):
+        transaction = get_transaction_from_id(pk)
+        submission = transaction.submission_price.submission
+
+        submission.generate_and_save_pdf("confirmation", transaction)
+
+        if (
+            not request.user == submission.author
+            or not transaction.status == transaction.STATUS_UNPAID
+        ):
+            raise PermissionDenied
+
+        processor = get_payment_processor(submission.get_form_for_payment())
+        if processor.is_transaction_authorized(transaction):
+            transaction.set_paid()
+            submission.prolongation_date = datetime.fromtimestamp(prolongation_date)
+            _set_prolongation_requested_and_notify(submission, request)
+            return render(
+                request,
+                "submissions/submission_payment_callback_confirm.html",
+                {
+                    "submission": submission,
+                },
+            )
+
+        transaction.set_failed()
```

Comment:
Not sure if it would be very useful, all of the additional relevant transactions information can be found the payment platform with their merchant IDs.